src/third-party/sonic/CMakeLists.txt

```diff
@@ -13,35 +13,60 @@
+set(Sonic_FORCE_COMPILATION CACHE BOOL "Force libsonic compilation")
 
-set_target_properties(sonic PROPERTIES VERSION "${LIBSONIC_VERSION}" SOVERSION "${LIBSONIC_VERSION_MAJOR}")
-set(CPACK_COMPONENT_LIBSONIC_VERSION_MAJOR "${LIBSONIC_VERSION_MAJOR}")
-set(CPACK_COMPONENT_LIBSONIC_VERSION_MINOR "${LIBSONIC_VERSION_MINOR}")
-set(CPACK_COMPONENT_LIBSONIC_VERSION_PATCH "${LIBSONIC_VERSION_PATCH}")
-set(CPACK_COMPONENT_LIBSONIC_VERSION "${LIBSONIC_VERSION}")
+set(DEB_LIBSONIC_PACKAGE "libsonic0")
+set(DEB_LIBSONIC_PACKAGES "${DEB_LIBSONIC_PACKAGE}, libsonic")
 
-target_include_directories(sonic PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}")
-harden(sonic)
-add_sanitizers(sonic)
+if(Sonic_FORCE_COMPILATION)
+	add_library(sonic SHARED "${SONIC_DIR}/sonic.c")
 
+	set(LIBSONIC_VERSION_MAJOR "0")
+	set(LIBSONIC_VERSION_MINOR "1")
+	set(LIBSONIC_VERSION_PATCH "17")
+	set(LIBSONIC_VERSION "${LIBSONIC_VERSION_MAJOR}.${LIBSONIC_VERSION_MINOR}.${LIBSONIC_VERSION_PATCH}")
 
-cpack_add_component(libsonic 
-	DISPLAY_NAME "libsonic"
-	DESCRIPTION "libsonic - a library for changing rate of speech. This component may contain a modified version of libsonic, for the modifications see the RHVoice GitHub repo. The original resides here: https://github.com/waywardgeek/sonic"
-	GROUP "third_party_libs"
-)
-list(APPEND CPACK_COMPONENTS_ALL "libsonic")
+	set_target_properties(sonic PROPERTIES VERSION "${LIBSONIC_VERSION}" SOVERSION "${LIBSONIC_VERSION_MAJOR}")
+	set(CPACK_COMPONENT_LIBSONIC_VERSION_MAJOR "${LIBSONIC_VERSION_MAJOR}")
+	set(CPACK_COMPONENT_LIBSONIC_VERSION_MINOR "${LIBSONIC_VERSION_MINOR}")
+	set(CPACK_COMPONENT_LIBSONIC_VERSION_PATCH "${LIBSONIC_VERSION_PATCH}")
+	set(CPACK_COMPONENT_LIBSONIC_VERSION "${LIBSONIC_VERSION}")
 
-install(TARGETS "sonic"
-	COMPONENT "libsonic"
-	LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
-)
-set(CPACK_DEBIAN_LIBSONIC_PACKAGE_NAME "libsonic")
+	target_include_directories(sonic PUBLIC "${SONIC_DIR}")
+	harden(sonic)
+	add_sanitizers(sonic)
+	set(CPACK_DEBIAN_LIBSONIC_PACKAGE_NAME "RHVoice-libsonic")
+	set(CPACK_DEBIAN_LIBSONIC_PACKAGE_REPLACES "${DEB_LIBSONIC_PACKAGES}")
+	set(CPACK_DEBIAN_LIBSONIC_PACKAGE_CONFLICTS "${CPACK_DEBIAN_LIBSONIC_PACKAGE_REPLACES}")
+	set(CPACK_DEBIAN_LIBSONIC_PACKAGE_BREAKS "${CPACK_DEBIAN_LIBSONIC_PACKAGE_REPLACES}")
+else()
+	add_library(sonic SHARED IMPORTED)
+	set_property(TARGET sonic PROPERTY IMPORTED_LOCATION "${Sonic_LIBRARIES}")
+	target_include_directories(sonic INTERFACE "${Sonic_INCLUDE_DIRS}")
+	set(CPACK_DEBIAN_LIBSONIC_PACKAGE_NAME "${DEB_LIBSONIC_PACKAGE}")
+endif()
+
+if(WIN32 OR Sonic_FORCE_COMPILATION)
+	cpack_add_component(libsonic 
+		DISPLAY_NAME "libsonic"
+		DESCRIPTION "libsonic - a library for changing rate of speech. This component may contain a modified version of libsonic, for the modifications see the RHVoice GitHub repo. The original resides here: https://github.com/waywardgeek/sonic"
```

Comment:
it is submodule so cannot contain any modifications.

Reply:
It is not, the submodule itself is in another dir.

Reply:
yes it isn't in third-party it is in: set(SONIC_DIR "${CMAKE_SOURCE_DIR}/external/libs/sonic")
but it cannot contain any modifications.

Reply:
I don't understand what you mean. `CMakeLists.txt` for sonic is in this repo and is not a part of `sonic` repo (though probably we should make a PR creating one in their repo doing proper packaging of libsonic, not the way it is curently done in Debian/Ubuntu (currently it is without pkg-config and CMake support (`FindSonic.cmake` should not be a part of this repo, instead it should be installed as a part of `libsonic-dev`)).

Reply:
ok now I see. but I still think the description is wrong.

> we should make a PR creating one in their repo doing proper packaging of libsonic, not the way it is curently done in Debian/Ubuntu 

yep. i want to use meson for that. i have WIP branch need to finish it.

Reply:
>i want to use meson for that. 

You may want to take a look at https://github.com/prebuilder/prebuilder.py . Also python-based and has more functionality for packaging. Unfortunately it has 3 flaws. 1. it got broken somewhen and I haven't yet fixed it. 2. it is not mature and not very finished  (i.e. the backends for anything except Debian are not really implemented, because I only use Ubuntu) and api-stable 3. It requires some changes in third-party software, such as dpkg, and these changes have not yet been accepted, they have even not been reviewed. I don't know what to do with that except either stating "one must use an our fork of apt" or maybe just extremely inefficiently working around the missing functionality.